From 7a8732a62b27485af85b51da4247ab59d9ef8537 Mon Sep 17 00:00:00 2001
From: sys_icgvert <icgvert@intel.com>
Date: Mon, 18 Apr 2016 18:45:20 -0700
Subject: [PATCH] atomisp2: css2400 and 2401 irci_ecr-master_20160418_1500

Merge CSS2.1 for css2400 and 2401_legacy pkg

Change-Id: Ie3f759b5c46e46e42c1d8c3701d9b9e517d4edf1
FW: tag irci_ecr-master_20160418_1500
Original-Tracked_On: #H1404950463
Signed-off-by: sys_icgvert <icgvert@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/114734
Approver: WeiX Meng <weix.meng@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: Dessert M BXT RVP (SC)
Build: Dessert M CHT FFD (SC)
Validation-Android: Dessert M BXT RVP (SC)
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-29585
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://android.intel.com:443/497641
---
 .../external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h | 2 +-
 .../camera/drivers/media/pci/atomisp2/css/ia_css_types.h            | 2 +-
 .../camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h     | 2 +-
 .../drivers/media/pci/atomisp2/css/runtime/debug/src/ia_css_debug.c | 6 ++++++
 .../external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css.c | 5 +++++
 .../camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c         | 4 ++--
 6 files changed, 16 insertions(+), 5 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
index b9a3119..7cf8804 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_ecr-master_20160325_1500 */
+/* Release Version: irci_ecr-master_20160418_1500 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
index c44f704..6fda5a6 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_ecr-master_20160325_1500 */
+/* Release Version: irci_ecr-master_20160418_1500 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
index e3adfc5..c34359d 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
@@ -22,7 +22,7 @@ more details.
 #define __IA_CSS_VERSION_DATA_H
 
 
-#define CSS_VERSION_STRING "REL:20160325_12.5_1933; API:2.1.22.2; GIT:irci___#35298f51b665ebcc2ebd2705f9bb92d7867be639; SDK:; USER:viedifw; "
+#define CSS_VERSION_STRING "REL:20160418_16.1_1833; API:2.1.22.2; GIT:irci___#2b04e084673172c9673146c645d49bab447c8aca; SDK:; USER:viedifw; "
 
 
 #endif
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/debug/src/ia_css_debug.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/debug/src/ia_css_debug.c
index a6f14bb..1ca949c 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/debug/src/ia_css_debug.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/debug/src/ia_css_debug.c
@@ -3327,12 +3327,18 @@ ia_css_debug_dump_stream_config(
 			config->pack_raw_pixels);
 	ia_css_debug_dtrace(IA_CSS_DEBUG_TRACE, "continuous: %d\n",
 			config->continuous);
+	ia_css_debug_dtrace(IA_CSS_DEBUG_TRACE, "disable_cont_viewfinder: %d\n",
+			config->disable_cont_viewfinder);
 	ia_css_debug_dtrace(IA_CSS_DEBUG_TRACE, "flash_gpio_pin: %d\n",
 			config->flash_gpio_pin);
 	ia_css_debug_dtrace(IA_CSS_DEBUG_TRACE, "left_padding: %d\n",
 			config->left_padding);
 	ia_css_debug_dump_mipi_buffer_config(&config->mipi_buffer_config);
 	ia_css_debug_dump_metadata_config(&config->metadata_config);
+	ia_css_debug_dtrace(IA_CSS_DEBUG_TRACE, "enable_raw_buffer_locking: %d\n",
+			config->ia_css_enable_raw_buffer_locking);
+	ia_css_debug_dtrace(IA_CSS_DEBUG_TRACE, "lock_all: %d\n",
+			config->lock_all);
 }
 
 /*
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css.c
index 7051315..d5807c1 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css.c
@@ -3612,6 +3612,11 @@ init_in_frameinfo_memory_defaults(struct ia_css_pipe *pipe,
 	in_frame->info.res.height = pipe->stream->config.input_config.input_res.height;
 	in_frame->info.raw_bit_depth =
 		ia_css_pipe_util_pipe_input_format_bpp(pipe);
+
+#if defined(IS_ISP_2500_SYSTEM)
+	in_frame->info.raw_type = pipe->stream->config.input_config.raw_type;
+#endif
+
 	ia_css_frame_info_set_width(&in_frame->info, pipe->stream->config.input_config.input_res.width, 0);
 	in_frame->contiguous = false;
 	in_frame->flash_state = IA_CSS_FRAME_FLASH_STATE_NONE;
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
index 74416e8..a27fb2c 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
@@ -47,10 +47,10 @@ struct fw_param {
 /* Warning: same order as SH_CSS_BINARY_ID_* */
 static struct firmware_header *firmware_header;
 
-/* The string STR(irci_ecr-master_20160325_1500) is a place holder
+/* The string STR(irci_ecr-master_20160418_1500) is a place holder
  * which will be replaced with the actual RELEASE_VERSION
  * during package generation. Please do not modify  */
-static const char *release_version = STR(irci_ecr-master_20160325_1500);
+static const char *release_version = STR(irci_ecr-master_20160418_1500);
 
 #define MAX_FW_REL_VER_NAME	300
 static char FW_rel_ver_name[MAX_FW_REL_VER_NAME] = "---";
-- 
1.9.1

